docs/ops/sort/MatrixNMS_8.md

```diff
@@ -0,0 +1,157 @@
+##  MatrixNms<a name="MatrixNms"></a> {#openvino_docs_ops_sort_MatrixNms_8}
+
+**Versioned name**: *MatrixNms-8*
+
+**Category**: *Sorting and maximization*
+
+**Short description**: *MatrixNms* performs matrix non maximum suppression (NMS) of the boxes with predicted scores.
+
+**Detailed description**: *MatrixNms* is a multi-phase operation.
+
+First it selects candidate bounding boxes that have higher scores than `score_threshold`. Then for each class at most `nms_top_k` number of candidate boxes are selected. Score of the candidate boxes are then decayed according to Matrix NMS algorithm [Wang et al](https://arxiv.org/abs/2003.10152.pdf).This algorithm is applied independently to each class and each batch element. Boxes of `background_class` are skipped and thus eliminated during the process. Finanly, it thresholds the decayed scores with `post_threshold`, and selects at most `keep_top_k` scoring candidate boxes per batch element.
+
+The Matrix NMS algorithm is described as below:
+1.  Compute `n x n` pairwise IoU matrix `X` for top N bounding boxes. Suppose `n` is the number of candidate boxes, which have been sorted descending by score, thresholded with `score_threshold` and then seleted with `nms_top_k`.
+2.  `triu`: Set the lower triangle and diagonal of `X` to 0.
+3.  Take th column-wise max of `X` to compute a vector `K` of maximun IoU for each candidate box.
+4.  Repeat element value of `K` along axis 1. Suppose this get the matrix `X_cmax`.
+5.  Compute the decay factor: `decay_factor = exp((X_cmax**2 - X**2) * gaussian_sigma)` if `decay_function` is `guassian`, else `decay_factor = (1 - X) / (1 - X_cmax)`.
+6.  Take the column-wise min of `decay_factor`, and element-wise multiply with scores to decay them.
+
+**Attributes**:
+
+* *sort_result*
+
+  * **Description**: *sort_result* specifiy order of output elements.
+  * **Range of values**: `class`, `score`, `none`
+    * *class* - sort selected boxes by class id (ascending) in each batch element.
+    * *score* - sort selected boxes by score (descending) in each batch element.
+    * *none* - do not guarantee the order in each batch element.
+  * **Type**: `string`
+  * **Default value**: `none`
+  * **Required**: *No*
+
+* *output_type*
+
+  * **Description**: the output tensor type
```

Comment:
It is not clear which output this attribute affects. For example, the first output with `[class_id, box_score, xmin, ymin, xmax, ymax]` cannot be of type `i64` and `i32` for sure. Please, clarify this and consider renaming the attribute to be more precise which output it affects.